### PR TITLE
Add asset-manager project

### PR DESCRIPTION
**Pull request by chrisroos**
We're updating the Asset Manager app to store and serve files from AWS
S3. This commit adds an asset-manager project to configure the S3 bucket
for file storage and the IAM user that can read/write files in that
bucket.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -98,7 +98,7 @@ end
 desc 'Configure the remote state location'
 task configure_s3_state: [:validate_environment, :purge_remote_state] do
   region      = 'eu-west-1'
-  bucket_name = "govuk-terraform-state-#{deploy_env}"
+  bucket_name = "cjr-govuk-terraform-state-#{deploy_env}"
```
Comment (surminus) [inline Rakefile]: Make sure we don't merge this in!

**Comment by surminus** [comment]
> How is this supposed to work with multiple TF_VAR_environment values?
I'd expected it to allow me to provision buckets/users in multiple
environments (test and integration, for example)

We use totally separate accounts for different "environments", so the GOV.UK Test account has a totally different set of credentials for the GOV.UK Integration account. The test account is just that, for testing so feel free to use that to deploy from your laptop etc. The GOV.UK Integration account should ideally be deployed using the [Jenkins job](https://deploy.integration.publishing.service.gov.uk/job/Deploy_Terraform_Project/). Same goes for Staging and Production.

> How do I get the credentials of the created IAM user stored in
environment variables on the target server?

Puppet. Create the environment variables and get the app to read them in, and then add the secret keys in our credentials store.

**Comment by chrisroos** [comment]
> We use totally separate accounts for different "environments", so the GOV.UK Test account has a totally different set of credentials for the GOV.UK Integration account. The test account is just that, for testing so feel free to use that to deploy from your laptop etc.

Where do I find the credentials for the test account, @surminus?

> Puppet. Create the environment variables and get the app to read them in, and then add the secret keys in our credentials store.

I should be OK to add the environment variables to Puppet but where/what is the "credentials store"?

**Comment by surminus** [comment]
1. You'll need a user adding. Someone with access will need to do this, so I'll create a Trello card on our backlog to do it.
2. The credentials store is in our deployment repo - you probably don't have access, but quite a few people do. When you have the Puppet ready we can look at doing this.

**Comment by chrisroos** [comment]
@surminus: I can see that the access and secret keys of my created IAM user end up in the terraform-asset-manager.tfstate file that's stored on S3. Do we manually copy them from there and paste them into the environment specific hieradata?

**Comment by chrisroos** [comment]
Hi @surminus. I've tidied this PR and am now ready for a review in order to get it merged. Is that something you can do or should I ask someone else?

**Comment by chrisroos** [comment]
Thanks @surminus. I'm going to get this merged.

**Pull request merged**